### ISSUE TITLE
Issue during Playwright e2e tests

### ISSUE THREAD

**Issue by marcinTichoniuk** [fingerprintjs/fingerprintjs#1039]
**- What's the scenario, what happens and what did you expect to happen?**

Fingerprint is adding spans and divs to the html body during e2e tests fired in Playwright what causes tests to fail. Also body tag is affected with additional styles. I want fingerprint to not add additional stuff to the body during e2e tests. 

<img width="621" alt="Zrzut ekranu 2024-09-17 o 10 25 20" src="https://github.com/user-attachments/assets/e5f5924e-078e-463b-9fe9-df474660fbc0">
<img width="571" alt="Zrzut ekranu 2024-09-24 o 13 50 06" src="https://github.com/user-attachments/assets/92ad032b-64d9-4306-a610-640b3079bec9">
<img width="1249" alt="Zrzut ekranu 2024-09-12 o 11 07 54" src="https://github.com/user-attachments/assets/a5c30868-2e62-418c-a381-1458fc518edc">
<img width="501" alt="Zrzut ekranu 2024-09-17 o 10 24 55" src="https://github.com/user-attachments/assets/ac9f01fb-749a-4264-aaf3-1b89e0d8451d">

**- What device and browser are you using?**

Problem exists on Playwright Chromium browsers, especially mobile. I didn't notice similar issue on Playwright Safari

**- What version of FingerprintJS are you using? (Bug reports not applicable to FingerprintJS master are subject to be closed without comment.)**

3.4.2

**Comment by Valve** [comment]
Hi @marcinTichoniuk 
Thanks for submitting this issue to us.
We'll take a look into it and get back to you.

**Comment by Finesse** [comment]
Hi @marcinTichoniuk. What you show is added inside an iframe which is injected into the page for a few milliseconds. FingerprintJS doesn't inject all the elements into the page itself. Currently FingerprintJS has 3 entropy sources that interact with the page DOM:
- [font_preferences.ts](https://github.com/fingerprintjs/fingerprintjs/blob/master/src/sources/font_preferences.ts) (what you show on the screenshots)
- [fonts.ts](https://github.com/fingerprintjs/fingerprintjs/blob/master/src/sources/fonts.ts)
- [dom_blockers.ts](https://github.com/fingerprintjs/fingerprintjs/blob/master/src/sources/dom_blockers.ts)

We can't stop FingerprintJS from adding additional stuff to the body, because they provide information used to calculate the fingerprint. What I can suggest:

1. Improve your tests to ignore or remove the elements added by FingerprintJS. We can consider adding predictable attributes to the elements added by FingerprintJS to help you finding them.
2. Don't use FingerprintJS in the e2e tests, or mock FignerprintJS.

**Comment by marcinTichoniuk** [comment]
Hi @Finesse, thank you for your response!

1. We've been trying to clean up after fingerprint, but it's tricky. Can't you make a clean up after calculating fingerprint and remove this stuff from the body? Is that possible?
2. The problem is that we're using external library which depends on fingerprint so it's a bit out of our control. 

We're using kind of a hack to bypass this problem in e2e tests, but it's not great and we would like to have more robust solution.

**Comment by Finesse** [comment]
> We've been trying to clean up after fingerprint, but it's tricky. Can't you make a clean up after calculating fingerprint and remove this stuff from the body? Is that possible?

FingerprintJS performs the cleanup immediately. Looks like your test goes inside the iframe. This is a temporary iframe created and removed by FingerprintJS, so you shouldn't go inside. What you show on your screenshots happens inside the iframe, not on your page, you page only has an iframe inserted for a few milliseconds.

**Comment by marcinTichoniuk** [comment]
Yes you do cleanup, but you just remove iframe, you don't make cleanup inside iframe - so if you add 4000px width to body it's still there. I think that Playwright is fired in iframe too and there is some kind of iframe conflict - that's why fingerprint iframe is called inside playwright page and fingerprint is adding some stuff to actual playwright page instead of to separate iframe.

**Comment by Finesse** [comment]
> fingerprint is adding some stuff to actual playwright page instead of to separate iframe

That sounds very weird, like somebody messed with the FingerprintJS code. What's the external library (which depends on FingerprintJS) that you use? If possible, please create and share a reproducible demo.

**Comment by marcinTichoniuk** [comment]
Hi @Finesse sorry for late response - we've been using https://www.npmjs.com/package/@rebilly/framepay-react.

**Comment by Finesse** [comment]
I've found out that the FingerprintJS code is a part of https://framepay.rebilly.com/framepay.js, but I see no suspicious changes to the FingerprintJS code in that file.

Please make a reproducible demo.
